From 64380644b0964020c9d89ecb97965f732cac0470 Mon Sep 17 00:00:00 2001
From: Nathan Moinvaziri <nathan@nathanm.com>
Date: Fri, 4 Jun 2021 15:24:23 -0700
Subject: [PATCH 01/13] Only need to add rem if it is greater than zero in
 chunkmemset.

---
 chunkset_tpl.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/chunkset_tpl.h b/chunkset_tpl.h
index 96f08d1..ae049e7 100644
--- a/chunkset_tpl.h
+++ b/chunkset_tpl.h
@@ -153,9 +153,10 @@ Z_INTERNAL uint8_t* CHUNKMEMSET(uint8_t *out, unsigned dist, unsigned len) {
     }
 
     /* Last, deal with the case when LEN is not a multiple of SZ. */
-    if (rem)
+    if (rem) {
         memcpy(out, from, rem);
-    out += rem;
+        out += rem;
+    }
 
     return out;
 }
-- 
2.32.0

