From 11d7321e04024a46a5438cdcf4ec740e019e4842 Mon Sep 17 00:00:00 2001
From: Nathan Moinvaziri <nathan@nathanm.com>
Date: Fri, 4 Jun 2021 15:27:26 -0700
Subject: [PATCH 02/13] Calculate from and out buffer advance only once in
 chunkcopy.

---
 chunkset_tpl.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/chunkset_tpl.h b/chunkset_tpl.h
index ae049e7..55e0708 100644
--- a/chunkset_tpl.h
+++ b/chunkset_tpl.h
@@ -19,11 +19,11 @@ Z_INTERNAL uint32_t CHUNKSIZE(void) {
    reliable. */
 Z_INTERNAL uint8_t* CHUNKCOPY(uint8_t *out, uint8_t const *from, unsigned len) {
     chunk_t chunk;
-    --len;
+    int32_t align = (--len % sizeof(chunk_t)) + 1;
     loadchunk(from, &chunk);
     storechunk(out, &chunk);
-    out += (len % sizeof(chunk_t)) + 1;
-    from += (len % sizeof(chunk_t)) + 1;
+    out += align;
+    from += align;
     len /= sizeof(chunk_t);
     while (len > 0) {
         loadchunk(from, &chunk);
-- 
2.32.0

