From 3a66bc74be2fbefaaebfaaec9de016c1bbc3ed01 Mon Sep 17 00:00:00 2001
From: Mika Lindqvist <postmaster@raasu.org>
Date: Tue, 22 Jun 2021 01:41:16 +0300
Subject: [PATCH 13/13] [functable] Add missing call to cpu_check_features().

---
 functable.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/functable.c b/functable.c
index acae446..32ab90e 100644
--- a/functable.c
+++ b/functable.c
@@ -241,6 +241,7 @@ Z_INTERNAL uint32_t adler32_stub(uint32_t adler, const unsigned char *buf, size_
 Z_INTERNAL uint32_t chunksize_stub(void) {
     // Initialize default
     functable.chunksize = &chunksize_c;
+    cpu_check_features();
 
 #ifdef X86_SSE2_CHUNKSET
 # if !defined(__x86_64__) && !defined(_M_X64) && !defined(X86_NOCHECK_SSE2)
-- 
2.32.0

